cumulus_library/studies/core/builder_encounter_coding.py

```diff
@@ -0,0 +1,123 @@
+        # this one
+        cols_with_data = []
+        for code_source in code_sources:
+            query = get_is_table_not_empty_query("encounter", code_source["name"])
+            cursor.execute(query)
+            progress.advance(task)
+
+            if len(cursor.fetchall()) > 0:
+                if code_source["is_array"]:
+                    query = get_is_table_not_empty_query(
+                        "encounter",
+                        "t1.row1",
+                        [
+                            {
+                                "source_col": code_source["name"],
+                                "table_alias": "t1",
+                                "row_alias": "row1",
+                            }
+                        ],
+                    )
+                else:
+                    query = get_is_table_not_empty_query(
+                        "encounter", f"{code_source['name']}.coding"
+                    )
+                cursor.execute(query)
+                progress.advance(task)
+
+                if len(cursor.fetchall()) > 0:
+                    if code_source["is_array"]:
+                        query = get_is_table_not_empty_query(
+                            "encounter",
+                            "t2.row2",
+                            [
+                                {
+                                    "source_col": code_source["name"],
+                                    "table_alias": "t1",
+                                    "row_alias": "row1",
+                                },
+                                {
+                                    "source_col": "row1.coding",
+                                    "table_alias": "t2",
+                                    "row_alias": "row2",
+                                },
+                            ],
+                        )
+                    else:
+                        query = get_is_table_not_empty_query(
+                            "encounter",
+                            "t1.row1",
+                            [
+                                {
+                                    "source_col": f"{code_source['name']}.coding",
+                                    "table_alias": "t1",
+                                    "row_alias": "row1",
+                                }
+                            ],
+                        )
+                    cursor.execute(query)
+                    progress.advance(task)
+                    if len(cursor.fetchall()) > 0:
+                        cols_with_data.append(code_source)
+                else:
+                    progress.advance(task)
+            else:
+                progress.advance(task)
+                progress.advance(task)
+        return cols_with_data
+
+    def prepare_queries(self, cursor: object, schema: str):
+        """Constructs queries related to condition codeableConcept
```

Comment:
copy/paste

Reply:
Why would ICD9/10 have anything to do with an Encounter code? 
That isn't the coding system (namespace) of encounters 

        config = CodeableConceptConfig(
            source_table="encounter",
            source_id="id",
            cc_columns=cols_with_data,
            target_table="core__encounter_coding",
            code_systems=[
                "http://snomed.info/sct",
                "http://hl7.org/fhir/sid/icd-10-cm",
                "http://hl7.org/fhir/sid/icd-9-cm",
            ],
        )

Reply:
Here is what we get from Cerner for Encounter CLASS system 

http://terminology.hl7.org/CodeSystem/v3-ActCode
http://terminology.hl7.org/CodeSystem/v2-0004

Reply:
for **servicetype**

Cerner says 
http://snomed.info/sct

and really should be 
http://hl7.org/fhir/ValueSet/service-type
http://terminology.hl7.org/CodeSystem/service-type

I'm not convinced though that we can expect to know the namespace (system) URL though.

Reply:
http://hl7.org/fhir/R4/valueset-service-type.html

Reply:
standard for **PRIORITY** is here, also not ICD9/10 

https://terminology.hl7.org/1.0.0//ValueSet-v3-ActPriority.html

Reply:
these systems are now being used in the codeableconcept generator